spec/rubocop/cop/rspec/capybara/specific_matcher_spec.rb

```diff
@@ -0,0 +1,335 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::RSpec::Capybara::SpecificMatcher do
+  it 'does not register an offense for `have_css` when elements with' \
+    'no matcher specified in detail' do
+    expect_no_offenses(<<-RUBY)
+      expect(page).to have_css('article')
+      expect(page).to have_css('body')
+      expect(page).to have_css('tbody')
+    RUBY
+  end
+
+  it 'does not register an offense for `have_css` when elements with' \
+    'not element' do
+    expect_no_offenses(<<-RUBY)
+      expect(page).to have_css('.a')
+      expect(page).to have_css('#button')
+      expect(page).to have_css('[table]')
+    RUBY
+  end
+
+  it 'does not register an offense for `have_no_css` when elements with' \
+  'no matcher specified in detail' do
+    expect_no_offenses(<<-RUBY)
+      expect(page).to have_no_css('article')
+      expect(page).to have_no_css('body')
+      expect(page).to have_no_css('tbody')
+    RUBY
+  end
+
+  it 'does not register an offense for `have_no_css` when elements with' \
+    'not element' do
+    expect_no_offenses(<<-RUBY)
+      expect(page).to have_no_css('.a')
+      expect(page).to have_no_css('#button')
+      expect(page).to have_no_css('[table]')
+    RUBY
+  end
+
+  it 'does not register an offense for `has_css` when elements with' \
+  'no matcher specified in detail' do
+    expect_no_offenses(<<-RUBY)
+      expect(page).to has_css('article')
+      expect(page).to has_css('body')
+      expect(page).to has_css('tbody')
+    RUBY
+  end
+
+  it 'does not register an offense for `has_css` when elements with' \
+    'not element' do
+    expect_no_offenses(<<-RUBY)
+      expect(page).to has_css('.a')
+      expect(page).to has_css('#button')
+      expect(page).to has_css('[table]')
+    RUBY
+  end
+
+  it 'does not register an offense for `has_no_css` when elements with' \
+  'no matcher specified in detail' do
+    expect_no_offenses(<<-RUBY)
+      expect(page).to has_no_css('article')
+      expect(page).to has_no_css('body')
+      expect(page).to has_no_css('tbody')
+    RUBY
+  end
+
+  it 'does not register an offense for `has_no_css` when elements with' \
+  'not element' do
+    expect_no_offenses(<<-RUBY)
+      expect(page).to has_no_css('.a')
+      expect(page).to has_no_css('#button')
+      expect(page).to has_no_css('[table]')
+    RUBY
+  end
+
+  it "registers an offense when using `have_css('button')`" do
+    expect_offense(<<-RUBY)
+    expect(page).to have_css('button')
+                    ^^^^^^^^ Use `have_button` instead of `have_css`.
```

Comment:
It feels that the offence relates to the whole matcher, not just the method name. It will probably have to change anyway for autocorrection.

Reply:
Indeed it is. I have modified it so that it is applied to the whole.